gems/aws-sdk-core/lib/aws-sdk-core/config_file.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+module AWS::SDK::Core
+  # Represents the configuration and credentials files as a single profile file.
+  # @api private
+  class ConfigFile
+    def initialize(config_profiles:, credentials_profiles:, sso_sessions:)
+      merge_files!(config_profiles, credentials_profiles)
+      @profiles = config_profiles.transform_values(&:to_h)
+      @sso_sessions = sso_sessions.transform_values(&:to_h)
+    end
+
+    attr_reader :profiles, :sso_sessions
+
+    # TODO: fill these out
```

Comment:
I'd argue for removing those from here - we have alternate ways of constructing credentials using the credentials chain and I think we should prefer that to this logic.